app/src/main/java/co/touchlab/kampkit/android/BreedViewModel.kt

```diff
@@ -19,6 +21,11 @@ class BreedViewModel : ViewModel() {
     )
 
     val breedStateFlow: StateFlow<DataState<ItemDataSummary>> = _breedStateFlow
+    private var currentJob: Job = Job()
```

Comment:
Can you explain the problem this is solving a little more? I'm not sure multiple operations at this level should be cancelling each other. I think this will lead to bugs, for example:
1. The user attempts to favorite two breeds very quickly. The second one cancels the first while it's still in flight
2. The user favorites a breed and then manually refreshes very quickly. The update is cancelled by the new get call.

Reply:
Yeah, in the new update, I'm no longer going to cancel favoriting a breed jobs, as it's a 1-shot coroutine, but will still cancel the Flows. 

As it is now, updating the breeds table also returns a new flow. If the current job isn't cancelled, then the lambda passed to the previous invocation of collect() is still called, and that same lambda passed to the current invocation of collect() is also called.

So there are a couple of ways around this. We can define 2 methods in BreedModel (our repository): 1 to observe the database as a Flow, and another to update the database with new records, so we only have the same Flow for the lifetime of the ViewModel. The other option is to cancel the previous Flow when collecting a new one. I don't want to cancel as soon as I get a non-loading response, though, because when favoriting a breed, I want that observable query, as a flow, to emit the updated data.

Reply:
The first method you suggested (separating observing from updating) sounds more favorable to me. I don't feel like we gain anything from entangling the observation with the refresh action. We can think of refresh (and initial load) as one of many possible user actions (intents?) that could change the state. One of the goals on the original architecture was a very clear source of truth for the view. We observed the db and that was it, it didn't have to care about how those updates happened. I think adding the DataState into that is important too, but hopefully we can keep the clear data flow